### ISSUE TITLE
Provide tutorials on how to run popular CNI solutions

### ISSUE THREAD

**Issue by lgfa29** [hashicorp/nomad#16460]
### Proposal

Nomad 0.12.0 added support for the Container Network Interface (CNI) specification and multiple vendors provide CNI plugins that, in theory, are supported by Nomad but, without more prescriptive guidance, users have struggled to deploy these solutions.

### Use-cases

Run popular CNI providers, such as Cilium, Calico etc. in Nomad

### Attempted Solutions

Some previous discussions in https://github.com/hashicorp/nomad/issues/12120 and https://github.com/hashicorp/nomad/issues/11748

> **Note**
> Issue extracted from https://github.com/hashicorp/nomad/issues/13824. Refer to that issue for more context.

**Comment by suikast42** [comment]
Indeed. Escpacially there is a bigt interesset in Cilium Nomad integration.

**Comment by brotherdust** [comment]
Cilium would be incredibly useful!

**Comment by thatsk** [comment]
why weave is not showing up its would be easy and much more friendly

**Comment by maxadamo** [comment]
> Cilium would be incredibly useful!

It's possible to use Cilium by Cosmonic on Nomad. However, you need to have a Consul cluster, dedicated to your Nomad cluster. 
For instance, I share one Consul cluster for all my environments (test, acceptance, production), and I use different Nomad cluster for test, acceptance and production. This is not going to work, because the key prefix of the cilium agent cannot be customized, and it will always write to the folder `/cilium`. 

This problem will be solve, if cilium agent can use the kvstore from Nomad.

**Comment by maxadamo** [comment]
> why weave is not showing up its would be easy and much more friendly

@thatsk most of the times, they need to be adapted. For instance for Cilium they had to create a custom application (called Netreap) which talks to Cilium. Maybe Weave needs to be reworked too.
